src/sandstorm/web-publishing.capnp

```diff
@@ -0,0 +1,81 @@
+# Sandstorm - Personal Cloud Sandbox
+# Copyright (c) 2015 Sandstorm Development Group, Inc. and contributors
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+@0xd5d3e63bd0a552b6;
+
+$import "/capnp/c++.capnp".namespace("sandstorm");
+
+using Util = import "util.capnp";
+using Handle = Util.Handle;
+
+interface WebSite {
+  # Represents an HTTP URL to which a web site can be published.
+  #
+  # Make a powerbox request for `WebSite` when you wish to publish a web site accessible over HTTP
+  # without going through the Sandstorm UI.
+
+  getUrl @0 () -> (path :Text);
+  # Get the URL of this resource.
+
+  struct Entity {
+    # An HTTP entity (using the term "entity" as defined in the HTTP spec). Think of this as the
+    # "payload" of an HTTP response.
+
+    mimeType @0 :Text;  # Content-Type header.
+    language @1 :Text;  # Content-Language header (optional).
+    encoding @2 :Text;  # Content-Encoding header (optional).
+
+    body :union {
+      bytes @3 :Data;
+      # Use when content is smaller than 1MB.
+
+      blob @4 :Util.Blob;
+      # Use when content is larger than 1MB. Currently, the blob must be created via uploadBlob().
```

Comment:
I suppose the idea here is that someday this blob could alternatively come from an object-storage grain?